x/wasm/internal/keeper/handler_plugin.go

```diff
@@ -19,32 +23,46 @@ type MessageHandler struct {
 }
 
 func NewMessageHandler(router sdk.Router, customEncoders *MessageEncoders) MessageHandler {
-	encoders := DefaultEncoders().Merge(customEncoders)
+	encoders := DefaultEncoders(nil, nil).Merge(customEncoders)
 	return MessageHandler{
 		router:   router,
 		encoders: encoders,
 	}
 }
 
-type BankEncoder func(sender sdk.AccAddress, msg *wasmTypes.BankMsg) ([]sdk.Msg, error)
+func NewMessageHandlerV2(router sdk.Router, channelKeeper types.ChannelKeeper, capabilityKeeper types.CapabilityKeeper, customEncoders *MessageEncoders) MessageHandler {
+	encoders := DefaultEncoders(channelKeeper, capabilityKeeper).Merge(customEncoders)
+	return MessageHandler{
+		router:   router,
+		encoders: encoders,
+	}
+}
+
+type BankEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.BankMsg) ([]sdk.Msg, error)
 type CustomEncoder func(sender sdk.AccAddress, msg json.RawMessage) ([]sdk.Msg, error)
-type StakingEncoder func(sender sdk.AccAddress, msg *wasmTypes.StakingMsg) ([]sdk.Msg, error)
-type WasmEncoder func(sender sdk.AccAddress, msg *wasmTypes.WasmMsg) ([]sdk.Msg, error)
+type StakingEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.StakingMsg) ([]sdk.Msg, error)
+type WasmEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.WasmMsg) ([]sdk.Msg, error)
+type IBCEncoder func(ctx sdk.Context, sender sdk.AccAddress, source cosmwasmv2.IBCEndpoint, msg *cosmwasmv2.IBCMsg) ([]sdk.Msg, error)
 
 type MessageEncoders struct {
 	Bank    BankEncoder
 	Custom  CustomEncoder
 	Staking StakingEncoder
 	Wasm    WasmEncoder
+	IBC     IBCEncoder
 }
 
-func DefaultEncoders() MessageEncoders {
-	return MessageEncoders{
+func DefaultEncoders(channelKeeper types.ChannelKeeper, capabilityKeeper types.CapabilityKeeper) MessageEncoders {
+	e := MessageEncoders{
 		Bank:    EncodeBankMsg,
 		Custom:  NoCustomMsg,
 		Staking: EncodeStakingMsg,
 		Wasm:    EncodeWasmMsg,
 	}
+	if channelKeeper != nil { // todo: quick hack to keep tests happy
```

Comment:
I like this, so it is only there and needed for the IBC tests and you don't have to rebuild everything